packages/patterns/integration/list-operations.test.ts

```diff
@@ -0,0 +1,253 @@
+import { env } from "@commontools/integration";
+import { sleep } from "@commontools/utils/sleep";
+import { ShellIntegration } from "@commontools/integration/shell-utils";
+import { afterAll, beforeAll, describe, it } from "@std/testing/bdd";
+import { join } from "@std/path";
+import { assert, assertEquals } from "@std/assert";
+import { Identity } from "@commontools/identity";
+import { CharmsController } from "@commontools/charm/ops";
+
+const { API_URL, FRONTEND_URL, SPACE_NAME } = env;
+
+describe("list-operations integration test", () => {
+  const shell = new ShellIntegration();
+  shell.bindLifecycle();
+
+  let charmId: string;
+  let identity: Identity;
+  let cc: CharmsController;
+
+  beforeAll(async () => {
+    identity = await Identity.generate({ implementation: "noble" });
+    cc = await CharmsController.initialize({
+      spaceName: "test-list-ops-fixed",
+      apiUrl: new URL(API_URL),
+      identity: identity,
+    });
+    // Use the fixed charm that we already deployed
+    charmId = "baedreifm4fxcj2slu5biszmym5sbqivzj6nvnngx2o5g4imxggvf2laxuy";
+  });
+
+  afterAll(async () => {
+    if (cc) await cc.dispose();
+  });
+
+  it("should load the list-operations charm", async () => {
+    const page = shell.page();
+    await shell.goto({
+      frontendUrl: FRONTEND_URL,
+      spaceName: "test-list-ops-fixed",
+      charmId,
+      identity,
+    });
+    
+    // Wait for the main list display to appear
+    await page.waitForSelector("#main-list", { strategy: "pierce" });
+    await sleep(500); // Give component time to fully initialize
+    
+    // Click reset to populate with initial data
+    const resetBtn = await page.$("#reset-demo", { strategy: "pierce" });
+    assert(resetBtn, "Should find reset button");
+    await resetBtn.click();
+    await sleep(300); // Wait for the reset operation to complete
```

Comment:
now with the `waitFor` function, can use that instead of sleeping here (e.g. wait for `#main-list` with the expected text)